.github/workflows/builds.yml

```diff
@@ -408,3 +408,43 @@ jobs:
           name: windows-installer
           path: .\build\installer-inno\bin
           if-no-files-found: error
+
+  freebsd-build:
+    name: freebsd-14-1
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Build on FreeBSD 14.1
+      id: test
+      uses: vmactions/freebsd-vm@v1
+      with:
+        usesh: true
+        prepare: |
+          pkg install -y \
+              cmake-core \
+              pkgconf \
+              git \
+              avahi-libdns \
+              curl \
+              libsm \
+              libxinerama \
+              libxrandr \
+              libxtst \
+              libxkbcommon \
+              glib \
+              ninja \
+              qt6-base \
+              qt6-tools \
+              qt6-declarative
+
+        run: |
+          git config --global --add safe.directory /home/runner/work/input-leap/input-leap
+          git submodule update --init --recursive
+          mkdir build
+          cd build
+          cmake -DCMAKE_BUILD_TYPE=Debug -DINPUTLEAP_BUILD_GUI=OFF -DINPUTLEAP_BUILD_TESTS=OFF -DINPUTLEAP_BUILD_INSTALLER=OFF ..
+          cmake --build .
```

Comment:
if you add a step using the upload action we can get access the build files later on 
It will look something like 
```
- uses: actions/upload-artifact@v4
        if: env.B_BUILD_TYPE != 'Release'
        with:
          name: ${{ matrix.name }}-debug
          path: ./InputLeap*.tar.gz 
          if-no-files-found: error
```
you will need to modify the path to reflect what ever package you make or archive you pack things into .

Reply:
I left out the build results upload on purpose. This is just an entry level build regression test, I very much doubt anyone ever needs the build results.